cdc/capture/capture.go

```diff
@@ -409,17 +417,26 @@ func (c *captureImpl) campaignOwner(ctx cdcContext.Context) error {
 			}
 			return errors.Trace(err)
 		}
-		// Campaign to be an owner, it blocks until it becomes the owner
-		if err := c.campaign(ctx); err != nil {
+
+		campaignCtx, cancel := context.WithTimeout(ctx, time.Second)
+		err = c.campaign(campaignCtx)
+		cancel()
+		if err != nil {
 			switch errors.Cause(err) {
 			case context.Canceled:
 				return nil
+			case context.DeadlineExceeded:
+				// last campaigned timeout, just retry
+				continue
 			case mvcc.ErrCompacted:
 				// the revision we requested is compacted, just retry
 				continue
 			}
-			log.Warn("campaign owner failed", zap.Error(err))
-			// if campaign owner failed, restart capture
+			if strings.Contains(err.Error(), "context deadline exceeded") {
+				continue
+			}
```

Comment:
No, it's borrowed from https://github.com/pingcap/tiflow/blob/a4d2b787d34e7917d3992621a6b7c6392e8f8c0a/cdc/sorter/unified/merger.go#L515

I met the error that does not match `context.DeadlineExceeded`, but can be handled by this error string match, it was from gRPC I think.

This logic is removed, since campaign in a timeout way is useless.